lms/djangoapps/courseware/access.py

```diff
@@ -801,6 +801,33 @@ def _visible_to_nonstaff_users(descriptor):
     return VisibilityError() if descriptor.visible_to_staff_only else ACCESS_GRANTED
 
 
+def _can_access_descriptor_with_milestones(user, descriptor, course_key):
+    """
+    Returns if the object is blocked by an unfulfilled milestone.
+
+    Args:
+        user: the user trying to access this content
+        descriptor: if not a sequential, access is always granted because only sequentials have milestones
+        course_key: key for the course for this descriptor
+    """
+    # only check milestones for sequentials, as gated content doesn't apply to other block types
```

Comment:
Is this true?  This may be true today, but the milestones framework was implemented generically so that the `content_id` could be for _any_ content.  Can we relax this constraint and not special case `sequential`?